icicle/src/fields/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+if (EXT_FIELD)
+  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -DEXT_FIELD")
+endif ()
+
+SET(SUPPORTED_FIELDS_WITHOUT_NTT grumpkin)
+
+set(TARGET icicle_field)
+
+set(SRC ${CMAKE_SOURCE_DIR}/src)
+
+set(FIELD_SOURCE ${SRC}/fields/field.cu)
+list(APPEND FIELD_SOURCE ${SRC}/vec_ops/extern.cu)
+list(APPEND FIELD_SOURCE ${SRC}/fields/mont.cu)
+
+# TODO: impl poseidon for small fields. note that it needs to be defined for the extension field!
+if (DEFINED CURVE)
```

Comment:
would icicle_field not compiled without those?

Reply:
poseidon rn is not implemented for babybear, so yeah, if we try building it, it will not compile